src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -1,38 +1,13 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-/**
- * Parameterize this service and add its implementation in a separate class.
- */
-public interface MachineService {
+public interface MachineService<T extends Machine> {
 
-    /**
-     * Return the list of machines.
-     * In the implementation of this method please use your MachineProducer implementations
-     * See that 'Class type' is not parametrized.
-     * Consider to parametrize this Class< PARAMETRIZE_ME > with specific wildcard bounding
-     *
-     * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
-     * @return the list of machines
-     */
-    List<Object> getAll(Class type);
+    List<T> getAll(Class<? extends Machine> type);
```

Comment:
```suggestion
    List<T> getAll(Class<? extends T> type);
```
в даному випадку краще використовувати тип `Т`, адже може бути ситуація що ми параметризовуємо наш MachineService наприклад бульдозером і відповідно очікуємо що наш сервіс буде працювати лише з бульдозерами і його наслідниками. Але якщо в методі ми будемо використовувати ` ? extends Machine` то це означає що в параметри методу ми зможемо передати все що наслідується від Machine, а це може бути Truck, Excavator і т.д, (хоча вони не є наслідниками бульдозера). Виходить логіка нашого класу буде не консистентною